frontend/javascripts/oxalis/view/left-border-tabs/modals/add_volume_layer_modal.tsx

```diff
@@ -14,9 +14,72 @@ import {
   getDatasetResolutionInfo,
   getSegmentationLayers,
 } from "oxalis/model/accessors/dataset_accessor";
-import { getVolumeTracingLayers } from "oxalis/model/accessors/volumetracing_accessor";
+import {
+  getAllReadableLayerNames,
+  getVolumeTracingLayers,
+} from "oxalis/model/accessors/volumetracing_accessor";
+import messages from "messages";
 import InputComponent from "oxalis/view/components/input_component";
 import api from "oxalis/api/internal_api";
+import Toast from "libs/toast";
+
+export type ValidationResult = { isValid: boolean; message: string };
+export function checkForLayerNameDuplication(
+  readableLayerName: string,
+  allReadableLayerNames: string[],
+): ValidationResult {
+  const layerNameDoesNotExist = !allReadableLayerNames.includes(readableLayerName);
+  return {
+    isValid: layerNameDoesNotExist,
+    message: layerNameDoesNotExist ? "" : messages["tracing.volume_layer_name_duplication"],
+  };
+}
+
+export function checkLayerNameForInvalidCharacters(readableLayerName: string): ValidationResult {
+  const uriSaveCharactersRegex = /[0-9a-zA-Z-._~]+/g;
+  // Removing all URISaveCharacters from readableLayerName. The left over chars are all invalid.
+  const allInvalidChars = readableLayerName.replace(uriSaveCharactersRegex, "");
+  const allUniqueInvalidCharsAsSet = new Set(allInvalidChars);
+  const allUniqueInvalidCharsAsString = "".concat(...allUniqueInvalidCharsAsSet.values());
+  const isValid = allUniqueInvalidCharsAsString.length === 0;
+  return {
+    isValid,
+    message: isValid
+      ? ""
+      : messages["tracing.volume_layer_name_includes_invalid_characters"](
+          allUniqueInvalidCharsAsString,
+        ),
+  };
+}
+
+export function validateReadableLayerName(
+  readableLayerName: string,
+  allReadableLayerNames: string[],
+  nameNotToCount?: string,
+): ValidationResult {
+  if (nameNotToCount) {
+    // If the given nameNotToCount is already once within the allReadableLayerNames we need to remove this name once from the allReadableLayerNames.
```

Comment:
```suggestion
    // nameNotToCount needs to be removed once if it is included in allReadableLayerNames.
```

Feel free to ignore, but sounds less clunky to me.